karabo_data/geometry2.py

```diff
@@ -205,6 +216,123 @@ def inspect(self, frontview=True):
+        return cls(modules, filename=filename)
+
+    def _get_rigid_groups(self, nquads=4):
+        """Create rigid stings for rigid groups definiton."""
+
+        quads = ','.join(['q{}'.format(q) for q in range(nquads)])
+        modules = ','.join(['p{}'.format(p) for p in range(self.n_modules)])
+
+        prod = product(range(self.n_modules), range(self.n_tiles_per_module))
+        rigid_group = ['p{}a{}'.format(p, a) for (p, a) in prod]
+        rigid_string = '\n'
+
+        for nn, rigid_group_q in enumerate(np.array_split(rigid_group, nquads)):
+            rigid_string += 'rigid_group_q{} = {}\n'.format(nn, ','.join(rigid_group_q))
+        rigid_string += '\n'
+        for nn, rigid_group_p in enumerate(np.array_split(rigid_group, self.n_modules)):
+            rigid_string += 'rigid_group_p{} = {}\n'.format(nn, ','.join(rigid_group_p))
+
+        rigid_string += '\n'
+
+        rigid_string += 'rigid_group_collection_quadrants = {}\n'.format(quads)
+        rigid_string += 'rigid_group_collection_asics = {}\n\n'.format(modules)
+        return rigid_string
+
+    def write_crystfel_geom(self, filename, *,
+                            data_path='/entry_1/instrument_1/detector_1/data',
+                            mask_path=None, extra_dim=None, adu_per_ev=None,
+                            clen=None, photon_energy=None):
+        """Write this geometry to a CrystFEL format (.geom) geometry file.
+        Parameters
+        ----------
+        filename : str
+        filename of the geometry file
+
+        data_path  : str, default : /entry_1/instrument_1/detector_1/data
+            path to the group that contains the data array in the hdf5 file
+        mask_path : str
+            path to the group that contains the mask array in the hdf5 file
+
+        extra_dim (dim_number, index) : collection
+            extra data dimension that has to inserted to read the data. The
+            first number of the tuple refers to the axis number; the second to
+            the index of that axis which holds the data of interest.
+            If for example raw data is to be read then extra_dim is could be
+            (2, 0).
+
+        adu_per_ev : float, default : 0.0075
+        ADU (analog digital units) per electron volt for the considered
+        detector.
+
+        clen : float, default : 0.119
+        Distance between sample and detector in meters
+
+        photon_energy : int, default : 9800
+        Beam wave length in eV
+        """
+        from . import __version__
+
+        if adu_per_ev is None:
+            adu_per_ev = 0.0075
+            warnings.warn('adu_per_ev must be set; Setting to default value')
+        if clen is None:
+            clen = 0.119
+            warnings.warn('clen must be set; Setting to default value')
+        if photon_energy is None:
+            photon_energy = 9800
+            warnings.warn('photon_energy must be set, setting to default value')
```

Comment:
I'd rather just write these commented out (`; photon_energy =`), or with a nonsense value that has to be replaced before you can use it with CrystFEL. People generally ignore warnings, especially if the warning isn't clear what needs to be done.

Reply:
Yes you're right! The comment idea (;) is a good one. This can also be useful once we offer beam_energy values as part of the virtual datasets and do not need to use the values defined here!

Reply:
I also guess adu_per_ev is detector specific, what about giving a default number per detector and falling back to this default value if not specified?

Reply:
If we can get good numbers and be confident that they're ~fixed for each detector, great. But unless you've got those numbers to hand already, I'd say leave it with no default for now, and we can work that out separately later.

Reply:
Yes that makes sense. But lets try not to forget about that. Maybe I'll add a TODO comment.